test/common/compressor/zlib_compressor_impl_test.cc

```diff
@@ -0,0 +1,128 @@
+#include <string>
+
+#include "envoy/common/exception.h"
+
+#include "common/buffer/buffer_impl.h"
+#include "common/common/assert.h"
+#include "common/common/hex.h"
+#include "common/compressor/zlib_compressor_impl.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Compressor {
+namespace {
+
+class ZlibCompressorImplTest : public testing::Test {
+protected:
+  static const int8_t gzip_window_bits{31};
+  static const int8_t memory_level{8};
+  static std::string multiply32BytesText(uint64_t n) {
+    std::string str{};
+    for (uint64_t i = 0; i < n; ++i) {
+      str.append("Sed ut perspiciatis unde omnis i");
+    }
+    return str;
+  }
+
+  static void compressorBadInitTestHelper(int8_t window_bits, int8_t mem_level) {
+    ZlibCompressorImpl compressor;
+    compressor.init(ZlibCompressorImpl::CompressionLevel::Standard,
+                    ZlibCompressorImpl::CompressionStrategy::Standard, window_bits, mem_level);
+  }
+
+  static void unitializedCompressorTestHelper() {
+    Buffer::OwnedImpl in;
+    Buffer::OwnedImpl out;
+    ZlibCompressorImpl compressor;
+    in.add(multiply32BytesText(1));
+    compressor.compress(in, out);
+  }
+};
+
+TEST_F(ZlibCompressorImplTest, CompressorTestDeath) {
```

Comment:
This should be called `ZlibCompressorImplDeathTest` (so that death tests run first, per https://github.com/google/googletest/blob/master/googletest/docs/AdvancedGuide.md#how-to-write-a-death-test)